### PR TITLE
updated ToS [SATURN-1351]

### PR DESCRIPTION
**Pull request by zarsky-broad**
On hold until the Mixpanel ToS updates are available.

TODO:
- [x] Tweak "as of" date before deploy
- [x] See if it's ok that I replaced the "cancel" button with "sign out" (since that's what it does)
- [x] Get the new version into the databases for alpha, perf, and staging (I've done the other envs already)
- [x] Accept new ToS as the test user on all envs so the integration tests pass OR teach the tests to handle this (done on dev already)

### CODE REVIEW

File: src/pages/TermsOfService.js [hunk 1]
```diff
@@ -1,281 +1,341 @@
...
-  constructor(props) {
-    super(props)
-    this.state = { busy: false }
-  }
+const TermsOfServicePage = () => {
+  const [busy, setBusy] = useState()
+  const { isSignedIn, acceptedTos } = authStore.get() // can't change while viewing this without causing it to unmount, so doesn't need to subscribe
 
-  async accept() {
+  const accept = async () => {
```
Comment (breilly2) [inline src/pages/TermsOfService.js:306]: Can/should this use `Utils.withBusyState`?

File: src/pages/TermsOfService.js [hunk 2]
```diff
@@ -1,281 +1,341 @@
...
+      div({ style: { maxHeight: 400, overflowY: 'auto', lineHeight: 1.5, marginTop: '1rem', paddingRight: '1rem' } }, [
+        h(MarkdownViewer, {
+          renderers: {
+            link: newWindowLinkRenderer,
+            heading: (text, level) => `<h${level} style="color: ${colors.dark()}; margin-bottom: 0">${text}</h${level}>`
+          }
+        }, [termsOfService])
+      ]),
+      isSignedIn && !acceptedTos && div({ style: { display: 'flex', justifyContent: 'flex-end', marginTop: '2rem' } }, [
+        h(ButtonSecondary, { style: { marginRight: '1rem' }, onClick: signOut }, 'Sign Out'),
```
Comment (breilly2) [inline src/pages/TermsOfService.js]: How about "Cancel and Sign Out"?

File: src/pages/TermsOfService.js [hunk 3]
```diff
@@ -1,281 +1,341 @@
...
-Terra uses cookies to enable sign on and other essential features when signed in, and to provide
-statistics to our development team regarding how the site is used. For more information, see our
-privacy policy. Privacy policy, including privacy policy of uploaded data, is [here](/#privacy).
-You agree to read and consider before uploading data.
+
+Terra uses cookies to enable sign on and other essential features when
+signed in, and to provide statistics to our development team regarding
+how the site is used. For more information, see our privacy policy.
+Privacy policy, including privacy policy of uploaded data, is
+[here](/#privacy). You agree to read and consider before uploading data.
```
Comment (breilly2) [inline src/pages/TermsOfService.js:42]: This probably isn't a new problem with this PR, but I just noticed that I can't view the privacy policy. It show briefly and then redisplays the Terms of Service.

File: src/pages/TermsOfService.js [hunk 4]
```diff
@@ -1,281 +1,341 @@
...
-      div({ style: styles.box }, [
-        termsTitle,
-        TOSMarkdown,
-        div({ style: { display: 'flex', justifyContent: 'flex-end', marginTop: '2rem' } }, [
-          h(ButtonSecondary, { style: { marginRight: '1rem' }, onClick: signOut }, 'Cancel'),
-          h(ButtonPrimary, { onClick: () => this.accept(), disabled: busy }, ['Accept'])
-        ])
+  return div({ style: { padding: '1rem', minHeight: '100%', display: 'flex', justifyContent: 'center', alignItems: 'center' } }, [
+    backgroundLogo,
+    div({ style: { backgroundColor: 'rgba(255,255,255,0.85)', borderRadius: 5, width: 800, padding: '2rem', boxShadow: Style.standardShadow } }, [
```
Comment (breilly2) [inline src/pages/TermsOfService.js]: We removed the opacity on the data import page at Joy's request. In the name of a11y, I suggest we do the same here.

File: src/pages/TermsOfService.js [hunk 5]
```diff
@@ -1,281 +1,341 @@
...
+      div({ style: { color: colors.dark(), fontWeight: 600 } }, [
+        span({ style: { fontSize: 36 } }, ['TERRA ']), span({ style: { fontSize: 24 } }, ['Terms of Service'])
```
Comment (breilly2) [inline src/pages/TermsOfService.js]: TOL: Strange that "TERRA" is all caps. 🤔

**Review by breilly2 — APPROVED** [review]
👍 Looks good overall. See comments.

I'm in favor of manually accepting the new TOS for our test account in each env instead of complicating the test code with conditional TOS acceptance... code which will be run so infrequently that it could even go stale and stop working by the next time we need it.

File: src/pages/TermsOfService.js (hunk 1 above)
Comment (zarsky-broad) [reply src/pages/TermsOfService.js:306]: That would cause a state change after unmount in the success case.

Comment (breilly2) [reply src/pages/TermsOfService.js:306]: Cool. At some point, could you teach me more about this? It's not quite clicking for me.

**Pull request merged**